File: src/service/OrderService.js
import service from './api';

class OrderService {

    search(search) {
        return service.post("/orders/search", search);
    }

    get(id) {
        return service.get(`/orders/${id}`);
    }

    create(data) {
        return service.post("/orders", data);
    }

    update(id, data) {
        return service.put(`/orders/${id}`, data);
    }

    delete(id) {
        return service.delete(`/orders/${id}`);
    }

    uploadFileExcel(objectFile) {
        return service.put(`/orders/upload-file`, objectFile, {
            headers: {
                'Content-type': 'multipart/form-data'
            }
        });
    }

    createOrderData() {
        return service.post("/orders/create-order-data");
    }

}

export default new OrderService();

File: back-end/vrp-preprocess/src/main/java/com/vrp/demo/service/OrderService.java
package com.vrp.demo.service;

import com.vrp.demo.entity.tenant.Order;
import com.vrp.demo.exception.CustomException;
import com.vrp.demo.models.OrderModel;
import com.vrp.demo.models.dashboard.SalesModels;
import com.vrp.demo.models.orders.OrderModelCreate;
import com.vrp.demo.models.search.OrderSearch;
import org.springframework.data.domain.Page;

import java.text.ParseException;
import java.util.List;

public interface OrderService extends BaseService<Order, Long> {

    public List<OrderModel> find(OrderSearch search);

    public Page<OrderModel> search(OrderSearch search);

    public OrderModel create(OrderModelCreate orderModelCreate) throws CustomException;
    public OrderModel create(OrderModel orderModel) throws CustomException;
    public OrderModel update(OrderModel orderModel) throws CustomException;

    public int delete(Long id) throws CustomException;

    public OrderModel findOne(Long id);

    public void createOrderData() throws CustomException;
    public List<SalesModels> searchByYear(String year) throws ParseException;
    public List<OrderModel> getOrdersByCustomer(Long userId);

}
